src/test/java/api/loans/agetolost/MarkOverdueLoansAsAgedLostServiceTest.java

```diff
@@ -0,0 +1,121 @@
+package api.loans.agetolost;
+
+import static api.support.APITestContext.getOkapiHeadersFromContext;
+import static api.support.matchers.ItemMatchers.isAgedToLost;
+import static api.support.matchers.ItemMatchers.isCheckedOut;
+import static org.folio.circulation.support.Clients.create;
+import static org.folio.circulation.support.http.OkapiHeader.OKAPI_URL;
+import static org.folio.circulation.support.http.OkapiHeader.REQUEST_ID;
+import static org.folio.circulation.support.http.OkapiHeader.TENANT;
+import static org.folio.circulation.support.http.OkapiHeader.TOKEN;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.joda.time.DateTime.now;
+import static org.joda.time.DateTimeZone.UTC;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.folio.circulation.services.agedtolost.MarkOverdueLoansAsAgedLostService;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.folio.circulation.support.http.server.WebContext;
+import org.joda.time.DateTime;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import api.support.builders.CheckOutByBarcodeRequestBuilder;
+import api.support.spring.SpringApiTest;
+import api.support.spring.clients.ScheduledJobClient;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.ext.web.RoutingContext;
+import lombok.val;
+
+public class MarkOverdueLoansAsAgedLostServiceTest extends SpringApiTest {
```

Comment:
Is this class testing the API or the internal service class?

If it is the latter, then I am glad that this can be tested in a smaller scoped tests. Could it be moved out from the API tests, or is it dependent upon the fake storage modules?

Reply:
In a fact it is an integration test, that requires Vert.x context and fake modules.

Reply:
It seems a little odd to me to have an integration test that encompasses many classes and an API test, as they are quite similar in both scope and intent.

What are the benefits of having both kind of test?

Could this be reworked to be more unit like?

Reply:
I have to add this test to test a specific test case, which unfortunately can not be verified using API test.

The test case is following: 
The process has been limited to process 1000 loans per single run, than next portion of loans will be picked up during the next run, and so on. I need to verify this behavior, that it works as expected. 

Using the default 1000 limit is quite time consuming for the API test. Because of this, I have created this integration test that overrides the limit and sets it to only 10 records.

Reply:
Thank you for explaining the reasoning behind this test.

Could that behaviour be demonstrated by a more unit like test that does not rely on the fake storage modules and other infrastructure?

For example, could that be demonstrated by extracting constructor parameters for the repositories and using a mock to verify that the correct limit is used?

Reply:
Hmm... I'd better remove it. In a fact this already covered by other tests: we verify that loans are not aged to lost twice, and that we can age to lost multiple loans.

Reply:
I wasn't really suggesting removing it. If this was a good way of demonstrating that the right number of records are processed and there isn't currently a better way to do that, then please feel free to leave it.